### ISSUE TITLE
MIssing newline in .min.js breaks downstream JS-combining builds

### ISSUE THREAD

**Issue by sgbeal** [jcubic/jquery.terminal#780]
### Issue summary

The distributed jquery.terminal.min.js has no newline on its final line. This leads to downstream pain for builds which combine multiple JS files into one, as the file appended after that one ends up being appended to a single-line comment:

```
//# sourceMappingURL=jquery.terminal.min.js.map/* This is the --pre-js file for emcc. It gets prepended to the
 ```

Yes, it can (must) be worked around downstream, but...

### Expected behavior

_All_ text files "really need" to end with a newline.

### Actual behavior

See above.

### Steps to reproduce

N/A

### Browser and OS

N/A

### Additional notes

N/A

**Comment by jcubic** [comment]
Thanks for the report. The file is generated by [Uglify-js](https://github.com/mishoo/UglifyJS) you can report this issue there. In the meantime, I can add a new line in Makefile.

**Comment by jcubic** [comment]
Can you share example of build script that fail to build because of the issue?

**Comment by sgbeal** [comment]
Makefile snippet (noting that the hard tab is replaced by spaces here):

```
bundle.js:
    cat jquery.terminal.min.js foo.js bar.js > $@
```

In my case, foo.js starts with the first line of a `/* ...comment`. The final line of the terminal script started with `// ...` and didn't have a newline, so that `/*...` was part of that comment, but the rest of the comment block was not.

It's trivial to work around, but annoying to have to work around it. It's _always_ good practice to have a final newline on text files, and some tools will complain if that's not the case.

**Comment by jcubic** [comment]
So you don't use any build system. So this is not a bug in UglifyJS. You can easily modify your script. If you would use Babel or any other builder instead of primitive Unix cat, everything would work as expected.

**Comment by sgbeal** [comment]
i use a makefile (a build system which predated node by 30 or 40 years ;)) to concatenate multiple JS files, one of which was copied verbatim from this project's git checkout:

```
[stephan@nuc:~/src/jquery.terminal/js]$ tail -1 jquery.terminal.min.js
//# sourceMappingURL=jquery.terminal.min.js.map[stephan@nuc:~/src/jquery.terminal/js]$ 
```

Note that the final line has no trailing newline, causing my shell prompt to continue on that same line. Whether that missing newline is caused by one of your own build tools or not, i have _no clue_, i just know that the file in question is part of your project, thus the bug report filed here.

Modifying my build was exactly what i ended up having to do, but not having a trailing newline is "poor practice" exactly because it causes grief for downstream tools. That's what i'm reporting.

**Comment by jcubic** [comment]
I'm not sure if you're aware. But the fix was released in version 2.33.3 it's already on NPM.

**Comment by jcubic** [comment]
BTW: I'm using `make` myself, you would look at the code you will see.
